Compiler/src/tfuncs.jl

```diff
@@ -1768,6 +1772,18 @@ const _tvarnames = Symbol[:_A, :_B, :_C, :_D, :_E, :_F, :_G, :_H, :_I, :_J, :_K,
         end
         return allconst ? Const(ty) : Type{ty}
     end
+    if 1 < unionsplitcost(𝕃, argtypes) ≤ max_union_splitting
+        ⊔ = join(𝕃)
+        rt = Bottom
+        for split_argtypes = switchtupleunion(𝕃, argtypes)
+            rt = rt ⊔ _apply_type_tfunc(𝕃, headtype, split_argtypes)
```

Comment:
Is this using `tmerge`? Generally only smerge is supposed to use that, while everything else tends to be supposed to use `Union{widenconst(...)...}`. I guess that might be a bigger fix than this PR though

Reply:
I thought the use of `tmerge` was to avoid constructing overly complex `Union`s, which helps prevent inference bloat, so its use is generally preferable? As long as `smerge` uses `tmerge`, the inference state shouldn’t bloat, so you’re probably right that there’s no need to use `tmerge` in other cases.